### DISCUSSION TITLE
Use readonly by default unless you save the file

### DISCUSSION THREAD

**Discussion by PerfectlyFineCode** [WerWolv/ImHex#1878] [Feedback]
Hi!

I've been using ImHex to read the save file of my program, however that also means ImHex locks the file so my own program can't access the file.

It would be a significantly better experience if it could open the file as readonly and you edit the file in-memory and when you save it, it will open the same file (or upgrade it) with write access.

That way there should be no issues with readonly / write access, but it will dynamically switch between them.

**Comment by WerWolv** [comment]
Hey, ImHex actually always opens files in a way that allows other programs to access the files freely. The issue is, other programs don't explicitly opt into this behaviour and want exclusive access so they fail to open the files.

With newer versions of ImHex now, opening a small enough file (smaller than 256MiB by default), ImHex will load the entire file into memory and watches for changes to the file made by other applications. This basically solves the issue for the most common case. For larger files, the file will still be mapped into memory and prevent applications that don't explicitly use `_fsopen` with the `_SH_DENYNO` flag from opening the file. There's nothing more I can do about this issue. It's the job of other applications to allow access to the file in this way.
